common/containers/Tabs/Swap/components/CurrencySwap.js

```diff
@@ -1,64 +1,44 @@
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
 import translate from 'translations';
-import { combineAndUpper } from 'api/bity';
-
-class CoinTypeDropDown extends Component {
-  constructor(props) {
-    super(props);
-  }
-
-  static propTypes = {
-    kind: PropTypes.any,
-    onChange: PropTypes.any,
-    kindOptions: PropTypes.any
-  };
-
-  render() {
-    return (
-      <span className="dropdown">
-        <select
-          value={this.props.kind}
-          className="btn btn-default"
-          onChange={this.props.onChange.bind(this)}
-        >
-          {this.props.kindOptions.map((obj, i) => {
-            return <option value={obj} key={i}>{obj}</option>;
-          })}
-        </select>
-      </span>
-    );
-  }
-}
+import { combineAndUpper } from 'utils/formatters';
+import * as swapTypes from 'actions/swapTypes';
+import SimpleDropDown from 'components/ui/SimpleDropdown';
+
+export type CurrencySwapReduxStateProps = {
+  bityRates: {},
+  originAmount: string | number,
+  destinationAmount: string | number,
+  originKind: string,
+  destinationKind: string,
+  destinationKindOptions: String[],
+  originKindOptions: String[]
+};
+
+export type CurrencySwapReduxActionProps = {
+  originKindSwap: (value: string) => swapTypes.OriginKindSwapAction,
+  destinationKindSwap: (value: string) => swapTypes.DestinationKindSwapAction,
+  originAmountSwap: (
+    value: string | number
+  ) => swapTypes.OriginAmountSwapAction,
+  destinationAmountSwap: (
+    value: string | number
+  ) => swapTypes.DestinationAmountSwapAction,
+  changeStepSwap: () => swapTypes.ChangeStepSwapAction
+};
 
 export default class CurrencySwap extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      disabled: false
-    };
-  }
+  props: CurrencySwapReduxStateProps & CurrencySwapReduxActionProps;
```

Comment:
Isn't `&` a bitwise operator? I tried to do this in the flow repl to see what would happen, and it didn't seem to like this.